metis/bin/metis_client

```diff
@@ -1222,45 +1222,135 @@ EOT
     end
   end
 
-  class Nuke < Command
-    usage "nuke <local_directory>"
+  class Checksum < Command
+    usage "checksum <local_directory>"
+
+    def parse(args)
+      options[:file] = "md5checksum.txt"
+      parse_opts(args) do |opts|
+        opts.on("-f", "--file FILE", "Recursive folder removal") do |file|
+          options[:file] = file
+        end
+      end
+    end
+
+    def md5_cache
+      @md5_cache ||= {}
+    end
+
+    def load_md5_cache(md5file)
+      md5dir = ::File.dirname(md5file)
+
+      File.readlines(md5file).each do |line|
+        checksum, file_name = line.chomp.split(/\s+/, 2)
+        full_path = Pathname.new(::File.join( md5dir, file_name )).cleanpath.to_s
+        next unless ::File.exists?(full_path)
+
+        md5_cache[full_path] = checksum
+      end
+    end
 
     def run(local_directory)
       # collect all files and folders
       folders, files = Dir[::File.join(local_directory, "**", "*")].partition { |f| ::File.directory?(f) }
 
       puts "Found #{files.count} files in #{folders.count} folders, total size #{byte_format(files.map{|f| ::File.size(f)}.sum)}."
 
-      deleted_files = deleted_folders = 0
-      remaining_size = 0
+      md5files = files.select do |file|
+        ::File.basename(file) == options[:file]
```

Comment:
Will this work if the user does not specify a `-f` option when invoking the command?

Reply:
Oh, nevermind, I see what you're doing! ... assumes you have the same "md5 checksum filename" in some subdirectories. What's not quite clear from the option description...